service/controller/key/key.go

```diff
@@ -1,15 +1,61 @@
 package key
 
 import (
+	"bytes"
+	"html/template"
+	"io"
+	"os"
+	"path/filepath"
+
 	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/ignition-operator/data"
 	"github.com/giantswarm/ignition-operator/pkg/label"
+	"github.com/giantswarm/ignition-operator/service/controller/controllercontext"
 	"github.com/giantswarm/microerror"
+	"github.com/shurcooL/httpfs/vfsutil"
+)
+
+const (
+	FilePath = "/files"
+	UnitPath = "/units"
 )
 
 func OperatorVersion(getter LabelsGetter) string {
 	return getter.GetLabels()[label.OperatorVersion]
 }
 
+func Render(spec controllercontext.ContextSpec, filesdir string) (map[string]string, error) {
+	files := make(map[string]string)
+
+	err := vfsutil.WalkFiles(data.Assets, filesdir, func(path string, f os.FileInfo, rs io.ReadSeeker, err error) error {
+		if f.Mode().IsRegular() {
+			file, err := vfsutil.ReadFile(data.Assets, path)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			tmpl, err := template.New(path).Parse(string(file))
+			if err != nil {
+				return microerror.Maskf(err, "failed to parse file %#q", path)
```

Comment:
You shouldn't mask arbitrary errors.

Reply:
Took this error handling out of k8scc. Removing masking.